util/dvsim/sim_utils.py

```diff
@@ -99,3 +100,116 @@ def vcs_cov_summary_table(buf):
+    returns a number if the job runtime was extracted successfully. The function will raise
+    appropriate exception if the extraction fails.
+    """
+    if tool == 'xcelium':
+        return xcelium_job_runtime(log_text)
+    elif tool == 'vcs':
+        return vcs_job_runtime(log_text)
+    else:
+        raise NotImplementedError(f"{tool} is unsupported for run times extraction.")
+
+
+def vcs_job_runtime(log_text: List) -> [float, str]:
+    """Capture and return the job_runtime from VCS simulator.
+
+    Example of VCS job_runtime: 'CPU Time:      1.210 seconds;'
+    """
+    # Reverse the log_text because the runtime is usually printed at the end of
+    # the file.
+    for line in reversed(log_text):
+        if "CPU Time" in line:
+            # Find pattern: CPU Time: "XXXX" seconds;.
+            m = re.search(r"(\d+\.\d+)", line)
+            if m:
+                return float(m.group(0)), "s"
+            else:
+                raise SyntaxError(f"Cannot find job runtime in line: " + line)
+    raise SyntaxError(f"Cannot find job runtime in this file")
+
+
+def xcelium_job_runtime(log_text: List) -> [float, str]:
+    """Capture and return the job_runtime from Xcelium simulator.
+
+    Example of Xcelium job_runtime:
+      'TOOL: xrun(64) 21.09-s006: Exiting on Jul 22, 2022 at 13:38:48 PDT (total: 00:00:02)'
+    """
+    # Reverse the log_text because the runtime is usually printed at the end of
+    # the file.
+    for line in reversed(log_text):
+        if "total:" in line and "TOOL" in line:
+            # Find pattern: TOOL .. (total: "XX":"XX":"XX")
+            m = re.search(r"total:.*(\d+):(\d+):(\d+)", line)
+            if m:
+                # Convert time format: HR:MIN:SEC to seconds.
+                val = re.findall(r"\d+", m.group(0))
+                job_runtime = (int(val[0]) * 60 + int(val[1])) * 60 + int(val[2])
+                return job_runtime, "s"
+            else:
+                raise SyntaxError(f"Cannot find job runtime in line: " + line)
+    raise SyntaxError(f"Cannot find job runtime in this file")
+
+
+def get_simulated_time(log_text: List, tool: str) -> [float, str]:
+    """Capture and return the job simulated_time and its unit.
+
+    The log file is passed as an input to the function, in addition to the tool used. The tool
+    returns a number if the simulated time was extracted successfully. The function will raise
+    appropriate exception if the extraction fails.
+    """
+    if tool == 'xcelium':
+        return xcelium_simulated_time(log_text)
+    elif tool == 'vcs':
+        return vcs_simulated_time(log_text)
+    else:
+        raise NotImplementedError(f"{tool} is unsupported for run times extraction.")
+
+
+def xcelium_simulated_time(log_text: List) -> [float, str]:
+    """Capture and return the simulated_time from Xcelium simulator.
+
+    Example of Xcelium simulated_time:
+      TODO: use non-UVM dependent printout once the PR is merged.
```

Comment:
Documented here once the PR is merged: https://github.com/lowRISC/opentitan/issues/13903